gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/api/use_case/RollbackApiUseCase.java

```diff
@@ -15,20 +15,158 @@
  */
 package io.gravitee.apim.core.api.use_case;
 
+import static java.util.stream.Collectors.toMap;
+import static java.util.stream.Collectors.toSet;
+
 import io.gravitee.apim.core.UseCase;
-import io.gravitee.apim.core.api.domain_service.RollbackApiDomainService;
+import io.gravitee.apim.core.api.crud_service.ApiCrudService;
+import io.gravitee.apim.core.api.domain_service.UpdateApiDomainService;
+import io.gravitee.apim.core.api.model.Api;
+import io.gravitee.apim.core.audit.domain_service.AuditDomainService;
+import io.gravitee.apim.core.audit.model.ApiAuditLogEntity;
 import io.gravitee.apim.core.audit.model.AuditInfo;
+import io.gravitee.apim.core.audit.model.event.ApiAuditEvent;
+import io.gravitee.apim.core.event.query_service.EventQueryService;
+import io.gravitee.apim.core.plan.domain_service.ClosePlanDomainService;
+import io.gravitee.apim.core.plan.domain_service.CreatePlanDomainService;
+import io.gravitee.apim.core.plan.domain_service.UpdatePlanDomainService;
+import io.gravitee.apim.core.plan.query_service.PlanQueryService;
+import io.gravitee.common.utils.TimeProvider;
+import io.gravitee.definition.model.DefinitionVersion;
+import io.gravitee.definition.model.v4.plan.Plan;
+import io.gravitee.definition.model.v4.plan.PlanStatus;
+import java.time.ZonedDateTime;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
 import lombok.RequiredArgsConstructor;
 
 @RequiredArgsConstructor
 @UseCase
 public class RollbackApiUseCase {
 
-    private final RollbackApiDomainService rollbackApiDomainService;
+    private final EventQueryService eventQueryService;
+    private final ApiCrudService apiCrudService;
+    private final UpdateApiDomainService updateApiDomainService;
+    private final PlanQueryService planQueryService;
+    private final CreatePlanDomainService createPlanDomainService;
+    private final UpdatePlanDomainService updatePlanDomainService;
+    private final ClosePlanDomainService closePlanDomainService;
+    private final AuditDomainService auditService;
 
     public void execute(Input input) {
-        rollbackApiDomainService.rollback(input.eventId(), input.auditInfo);
+        Api api = eventQueryService
+            .findPublishApiPayload(input.eventId)
```

Comment:
As you return an Api, I would name it `findPublishedApi`

Reply:
it is just a detail, but I would have do it in a dedicated ApiEventQueryService, as it mixes the concept of Event and Api, but it can be refactored later.